process/block/preprocess/sovereignChainTransactions_test.go

```diff
@@ -608,4 +608,191 @@ func TestTxsPreprocessor_IsTransactionEligibleForExecutionShouldWork(t *testing.
 		assert.Equal(t, uint64(3), nonce)
 		assert.Equal(t, big.NewInt(1), balance)
 	})
+
+	t.Run("isTransactionEligibleForExecution should return false when relayer account is nil", func(t *testing.T) {
+		t.Parallel()
+
+		args := createDefaultTransactionsProcessorArgs()
+		args.TxProcessor = &testscommon.TxProcessorMock{
+			GetSenderAndReceiverAccountsCalled: func(tx *transaction.Transaction) (state2.UserAccountHandler, state2.UserAccountHandler, error) {
+				return &state.UserAccountStub{
+					Nonce:   1,
+					Balance: big.NewInt(10),
+				}, nil, nil
+			},
+			GetRelayerAccountCalled: func(tx *transaction.Transaction) (state2.UserAccountHandler, error) {
+				return nil, nil
+			},
+		}
+
+		tp, _ := NewTransactionPreprocessor(args)
+		sctp, _ := NewSovereignChainTransactionPreprocessor(tp)
+
+		tx := &transaction.Transaction{
+			SndAddr:          []byte("X"),
+			Nonce:            1,
+			Value:            big.NewInt(2),
+			RelayerAddr:      []byte("R"),
+			RelayerSignature: []byte("RS"),
+		}
+		err, value := sctp.isTransactionEligibleForExecution(tx, nil)
+		require.Nil(t, err)
+		require.False(t, value)
+	})
+
+	t.Run("isTransactionEligibleForExecution should return false if relayer doesn't have sufficient funds for fee", func(t *testing.T) {
```

Comment:
How come fee =10 and balance = 10, but this still returns not enough funds?

Reply:
fee = 10
sender balance = 10
relayer balance = 1
=> relayer doesn't have enough balance to pay the fee